roles/testnode/README.rst

```diff
@@ -0,0 +1,215 @@
+
+The host to use as a github mirror::
+
+    git_mirror_host: git.ceph.com
+
+This host is used by teuthology to download ceph packages and will be given higher priority on apt systems::
+
+    gitbuilder_host: gitbuilder.ceph.com
+
+The mirror to download and install ``pip`` from::
+
+    pip_mirror_url: "http://{{ mirror_host }}/pypi/simple"
+
+A hash defining yum repos that would be common across a major version. Each key in the hash will be
+the name of a yum repo created in /etc/yum.repos.d. The key/value pairs as the value for that repo
+will be used as the properties for the repo file::
+
+    common_yum_repos: {}
+
+    # An example: 
+    common_yum_repos:
+      rhel-7-fcgi-ceph:
+        name: "RHEL 7 Local fastcgi Repo"
+        baseurl: http://gitbuilder.ceph.com/mod_fastcgi-rpm-rhel7-x86_64-basic/ref/master/
+        enabled: 1
+        gpgcheck: 0
+        priority: 2
+
+A hash defining version specific yum_repos. Each key in the hash will be
+the name of a yum repo created in /etc/yum.repos.d. The key/value pairs as the value for that repo
+will be used as the properties for the repo file::
+
+    yum_repos: {}
+    
+    # An example:
+    yum_repos:
+      fedora-fcgi-ceph:
+        name: Fedora Local fastcgi Repo
+        baseurl: http://gitbuilder.ceph.com/mod_fastcgi-rpm-fedora20-x86_64-basic/ref/master/
+        enabled: 1
+        gpgcheck: 0
+        priority: 0
+
+A list of packages to install that is specific to a distro version.  These lists are defined in the var files in ``vars/``::
+
+    packages: []
+
+A list of packages to install that are common to a distro or distro version. These lists are defined in the var files in ``vars/``::
+
+    common_packages: []
+
+A list of packages to remove. These lists are defined in the var files in ``vars/``::
+
+    packages_to_remove: []
+
+A list of packages to upgrade. These lists are defined in the vars files in ``vars/``::
+
+    packages_to_upgrade: []
+
+The user that teuthology will use to connect to testnodes. This user will be created by this role and assigned to the appropriate groups.
+Eventhough this variable exists, teuthology is not quite ready to support a configurable user::
+
+    teuthology_user: "ubuntu"
+
+This user is created for use in running xfstests from ceph-qa-suite::
+
+    xfstests_user: "fsgqa"
+
+This will control whether or not rpcbind is started before nfs.  Some distros require this, others don't::
+
+    start_rpcbind: true
+
+Set to true if /etc/fstab must be modified to persist things like mount options, which is useful for long lived
+bare metal machines, less useful for virtual machines that are re-imaged before each job::
+
+    modify_fstab: true
+
+A list of ntp servers to use::
+
+    ntp_servers:
+      - 0.us.pool.ntp.org
+      - 1.us.pool.ntp.org
+      - 2.us.pool.ntp.org
+      - 3.us.pool.ntp.org
+
+The lab domain to strip from hostname on rhel 7.x nodes. This is only done if ``lab_domain``
```

Comment:
I wish I knew, I suspect it had something to do with how we used to cobbler rhel nodes but don't know for certain.  This is one of the ceph-qa-chef quirks we need to revisit eventually.